Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,195 @@
+# Non-Interactive Onchain Protocol
+
+# Table of Contents
+
+* [Abstract](#abstract)
+* [General Requirements](#general-requirements)
+* [Funding Phase](#funding-phase)
+* [Execution Phase](#execution-phase)
+* [Fee-Bumping Policy](#fee-bumping-policy)
+* [Reorgs](#reorgs)
+
+## Abstract
+
+DLC allows two parties to conduct stateful bitcoin contracts by giving each
+of the parties *cross-signed range of CET transactions*. These CETs transactions
+are all spending the same output from a funding transaction and thus only one of
+CET can confirm. A CET is finalized according to the outcome signed by the previously
+selected oracle.
+
+As the contract is established once funding transaction signatures have been exchanged,
+from then a party should assume counterparty non-cooperation and ensure either
+confirmation of the funding or double-spend of its collatral inputs to avoid exploitation.
+
+Once funding transaction signatures are exchanged, DLC parties aren't required
+to interact anymore. They are three scenarios to consider:
+
+1. The funding transaction doesn't confirm, a party must either bump its feerate
+to confirm before DLC maturing or double-spend its funding collateral, thus cancelling the
+DLC.
+
+2.  The funding transaction confirms, the oracle releases the outcome signature, a
+participant must broadcast the DLC transaction paired with the outcome.
+
+3. The funding transaction confirms, oracle doesn't release the outcome signature, independently
+of counterparty behavior, the party must broadcast the refund transaction after its timelock
+expiration.
+
+# General Requirements
+
+A DLC client should have censorship-resistant access to the blockchain, use a
+local fee-estimator and ensure it's well-connected to the tx-relay network.
+
+We recommend to not react on any mempool event as otherwise it would be a source of
+cross-layer mapping. <sup>[Cross-layer Mapping](https://arxiv.org/pdf/2007.00764.pdf)</sup>
+
+We recommend to not sign transactions until their broadcast is required
+to avoid their misuage in case of site compromise, assuming an external signer.
+
+# Funding Phase
+
+A party always starts the protocol in the funding phase. The funding phase can
+terminate by either a transition to the execution phase or a protocol abort.
+
+## Requirements
+
+A node:
+	- if a counterparty's funding signature is received:
+		- MUST finalize the funding transaction with a local `SIGHASH_ALL` signature
```

Comment:
Here and below funding signature -> funding signatures (as there can be many inputs with many sigs)